### PR TITLE
[WIP] ifqld with static organization+bucket

### PR DESCRIPTION
**Pull request by aanthony1243**

### CODE REVIEW

File: cmd/ifqld/main.go [hunk 1]
```diff
@@ -103,16 +110,37 @@ func ifqlF(cmd *cobra.Command, args []string) {
 	}
 }
 
+func getHosts(key string) ([]string, error) {
+	v := viper.GetViper()
+	hostStr := v.GetString(key)
+	if hostStr == "" {
+		return nil, errors.New("empty organization host string")
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: This should return a more generic message.

File: cmd/ifqld/main.go [hunk 2]
```diff
@@ -84,14 +87,18 @@ func ifqlF(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	var orgSvc platform.OrganizationService
 	// TODO(adam): figure out what orgSvc we need to create here.
+	orgHost, err := getHosts("ORGANIZATION_HOSTS")
+	if err != nil {
+		return
+	}
+	orgSvc := http.OrganizationService{Addr: orgHost[0]}
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: We do not want the Organization and Bucket services to be external dependencies of this binary.

We should just create a mock implementation of the Bucket and Organization services that has a single hardcoded org and bucket.

File: cmd/ifqld/main.go [hunk 3]
```diff
@@ -155,3 +180,101 @@ func (b bucketLookup) Lookup(orgID id.ID, name string) (id.ID, bool) {
...
+	if n < 1 {
+		return nil, fmt.Errorf("expected at least one organization")
+	}
+
+	return os[0], nil
+}
+
+func (s *StaticOrganizationService) FindOrganizations(ctx context.Context, filter platform.OrganizationFilter, opt ...platform.FindOptions) ([]*platform.Organization, int, error) {
+	po := platform.Organization{
+		ID:   platform.ID("staticorgid"),
```
[the remark this answers is not in the document]
Comment (aanthony1243) [reply cmd/ifqld/main.go]: don't know.  worked in tests...but I never really inspected it.

File: cmd/ifqld/main.go [hunk 4]
```diff
@@ -53,13 +56,13 @@ func init() {
...
+	ifqlCmd.PersistentFlags().String("bucket-name", "", "The bucket to access. ")
+	viper.BindEnv("BUCKET_NAME")
+	viper.BindPFlag("BUCKET_NAME", ifqlCmd.PersistentFlags().Lookup("bucket-name"))
 
-	ifqlCmd.PersistentFlags().String("organization-hosts", "", "The organization service host.")
-	viper.BindEnv("ORGANIZATION_HOSTS")
-	viper.BindPFlag("ORGANIZATION_HOSTS", ifqlCmd.PersistentFlags().Lookup("organization-hosts"))
+	ifqlCmd.PersistentFlags().String("organization-name", "", "The organization name to use.")
+	viper.BindEnv("ORGANIZATION_NAME")
+	viper.BindPFlag("ORGANIZATION_NAME", ifqlCmd.PersistentFlags().Lookup("organization-name"))
```
[the remark this answers is not in the document]
Comment (aanthony1243) [reply cmd/ifqld/main.go]: I thought that it'd provide some flexibility to at least rename the org + bucket at startup

[the remark this answers is not in the document]
Comment (nathanielc) [reply cmd/ifqld/main.go]: Yeah, but maybe I missed it below, but the names `staticbucket` and `staticorg` are hardcoded, not using these env vars.

File: cmd/ifqld/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathanielc) [reply cmd/ifqld/main.go]: I think I know whats up. The ID type is a `[]byte` so any string can be type cast to it. But not any string is valid. Maybe we should use the `DecodeFromString` method to ensure we have a valid type.

File: cmd/ifqld/main.go [hunk 5]
```diff
@@ -155,3 +180,101 @@ func (b bucketLookup) Lookup(orgID id.ID, name string) (id.ID, bool) {
...
+		return nil, fmt.Errorf("expected at least one organization")
+	}
+
+	return os[0], nil
+}
+
+func (s *StaticOrganizationService) FindOrganizations(ctx context.Context, filter platform.OrganizationFilter, opt ...platform.FindOptions) ([]*platform.Organization, int, error) {
+	po := platform.Organization{
+		ID:   platform.ID("staticorgid"),
+		Name: "staticorganization",
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: Should this not use s.Name. And the default be specified in flags default?

Comment (aanthony1243) [reply cmd/ifqld/main.go]: yes...

**Comment by pauldix** [comment]
The bucket name doesn't map to `db/rp`?

**Pull request merged**